spring-cloud-dataflow-configuration-metadata/src/main/java/org/springframework/cloud/dataflow/configuration/metadata/BootApplicationConfigurationMetadataResolver.java

```diff
@@ -106,29 +97,18 @@ public boolean supports(Resource app) {
 	 *            but directories are supported as well
 	 */
 	public List<ConfigurationMetadataProperty> listProperties(Resource app, boolean exhaustive) {
-		File moduleFile = null;
 		try {
-			moduleFile = app.getFile();
-		}
-		catch (IOException e) {
-			return Collections.emptyList();
-		}
-		Archive archive = null;
-		try {
-			archive = moduleFile.isDirectory() ? new ExplodedArchive(moduleFile) : new JarFileArchive(moduleFile);
+			Archive archive = resolveAsArchive(app);
 			return listProperties(archive, exhaustive);
 		}
 		catch (IOException e) {
-			throw new RuntimeException("");
+			throw new RuntimeException("Failed to list properties for " + app, e);
 		}
 	}
 
 	public List<ConfigurationMetadataProperty> listProperties(Archive archive, boolean exhaustive) {
-		ClassLoader moduleClassLoader = null;
-		try {
-
+		try (URLClassLoader moduleClassLoader = new BootClassLoaderFactory(archive, parent).createClassLoader()) {
```

Comment:
Since the parent class loader is always 'null' and no longer needed to use any of parent classloaders to load the metadata resources, can we simplify the `BootClassLoaderFactory` as well as `BootApplicationConfigurationMetadataResolver` not to use the `parent` class loader at all? and just use the 'null' option when creating `LaunchedURLClassLoader`?